components/dispatcher/internal/dispatcher/dispatcher_test.go

```diff
@@ -93,18 +101,46 @@ func TestDispatcher(t *testing.T) {
 		testBrokers,
 		testTopic,
 		testOffset,
-		testConcurrency,
-		testGroupId,
 		testOffsetCommitCount,
 		testOffsetCommitDuration,
 		testUsername,
-		testPassword)
-
-	// Start Consumers
-	testDispatcher.StartConsumers()
+		testPassword,
+		testChannelKey)
+
+	// Start 1 Consumer
+	testDispatcher.UpdateSubscriptions([]Subscription{
+		{
+			URI:     testSubscriberUri1,
+			GroupId: testGroupId,
+		},
+	})
 
 	// Verify Consumers Started
-	verifyConsumersCount(t, testDispatcher.consumers)
+	verifyConsumersCount(t, testDispatcher.consumers, 1)
+
+	// Remove All Consumers
+	testDispatcher.UpdateSubscriptions(make([]Subscription, 0))
+
```

Comment:
all these updateSubscriptions() calls should have an assert that returned err is nil?  Unless we remove the error response from the function as called for in another PR comment?